src/olympia/accounts/views.py

```diff
@@ -385,12 +390,39 @@ def logout_user(request, response):
         API_TOKEN_COOKIE, domain=settings.SESSION_COOKIE_DOMAIN)
 
 
+# This view is not covered by the CORS middleware, see:
+# https://github.com/mozilla/addons-server/issues/11100
 class SessionView(APIView):
     permission_classes = [IsAuthenticated]
 
+    def options(self, request, *args, **kwargs):
+        response = Response()
+        response['Content-Length'] = '0'
+        origin = request.META.get('HTTP_ORIGIN')
+        if not origin:
+            return response
+        response[ACCESS_CONTROL_ALLOW_ORIGIN] = origin
+        response[ACCESS_CONTROL_ALLOW_CREDENTIALS] = 'true'
+        # Mimics the django-cors-headers middleware.
```

Comment:
Re-implementing parts of `django-cors-headers` makes me wonder if we should consider an alternate approach. What about doing the logout as a redirect, as mentioned in https://github.com/mozilla/addons-server/issues/11100#issuecomment-478919885 ? That seems like a reasonable solution.

Reply:
I don't think that won't be a big problem given it's only for this view and only for logout and as such should only be happening once in the whole code.